Swiftfin tvOS/Views/UserSignInView/Components/PublicUserButton.swift

```diff
@@ -41,39 +50,49 @@ extension UserSignInView {
             .aspectRatio(1, contentMode: .fill)
         }
 
+        // MARK: - Person View
+
+        @ViewBuilder
+        private var personView: some View {
+            ZStack {
+                Color.clear
+
+                ImageView(user.profileImageSource(client: client, maxWidth: 120))
+                    .image { image in
+                        image
+                            .posterBorder(ratio: 0.5, of: \.width)
+                    }
+                    .placeholder { _ in
+                        fallbackPersonView
+                    }
+                    .failure {
+                        fallbackPersonView
+                    }
+            }
+        }
+
+        // MARK: - Body
+
         var body: some View {
             Button {
```

Comment:
I've recently checked out the new additions in tvOS since there has been more traction in this area. A lot has been added to make better buttons that we will adopt for the platform.

Changing the button to below will get us a better focus effect and will automatically move the labels on hover:

```swift
Button(action: action) {
    personView
        .aspectRatio(1, contentMode: .fill)
        .posterShadow()
        .clipShape(.circle)
        .frame(width: 150, height: 150)
        .hoverEffect(.highlight)

    Text(user.name ?? .emptyDash)
        .fontWeight(.semibold)
        .foregroundStyle(.primary)
        .lineLimit(1)
        .padding(.bottom)
}
.buttonBorderShape(.circle)
.buttonStyle(.borderless)
.disabled(!isEnabled)
.foregroundStyle(.primary)
```

I would recommend looking at the following sample project to see the new additions that we will be referencing.

- https://developer.apple.com/documentation/swiftui/creating-a-tvos-media-catalog-app-in-swiftui

Reply:
What!? That looks so much better! Big fan of how this looks!

![Simulator Screenshot - Apple TV 4K (3rd generation) - 2024-12-17 at 16 26 20](https://github.com/user-attachments/assets/60769ee8-7bdb-48b3-b5e5-bec475612cf0)